tests/model/model.cpp

```diff
@@ -624,3 +624,491 @@ TEST(Model, removeComponentInsensitiveToOrder)
+    EXPECT_TRUE(model->replaceUnits("myConcreteUnits1", myImportedUnits1));
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(2), model->importSourceCount());
+    EXPECT_EQ(import1, model->importSource(1));
+
+    // REPLACE imported -> concrete by name.
+    EXPECT_TRUE(model->replaceUnits("myImportedUnits2", myConcreteUnits2));
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(2), model->importSourceCount());
+}
+
+TEST(Model, replaceImportedUnitsByIndex)
+{
+    auto model = libcellml::Model::create();
+    auto myConcreteUnits1 = libcellml::Units::create("myConcreteUnits1");
+    auto myImportedUnits1 = libcellml::Units::create("myImportedUnits1");
+    auto myConcreteUnits2 = libcellml::Units::create("myConcreteUnits2");
+    auto myImportedUnits2 = libcellml::Units::create("myImportedUnits2");
+
+    auto import1 = libcellml::ImportSource::create();
+    auto import2 = libcellml::ImportSource::create();
+
+    import1->setUrl("import1.cellml");
+    import2->setUrl("import2.cellml");
+
+    myImportedUnits1->setImportSource(import1);
+    myImportedUnits2->setImportSource(import2);
+
+    model->addUnits(myConcreteUnits1);
+    model->addUnits(myImportedUnits2);
+
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+
+    // REPLACE concrete -> imported by index.
+    EXPECT_TRUE(model->replaceUnits(0, myImportedUnits1));
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(2), model->importSourceCount());
+    EXPECT_EQ(import1, model->importSource(1));
+
+    // REPLACE imported -> concrete by index.
+    EXPECT_TRUE(model->replaceUnits(1, myConcreteUnits2));
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(2), model->importSourceCount());
+}
+
+TEST(Model, replaceImportedUnitsByReference)
+{
+    auto model = libcellml::Model::create();
+    auto myConcreteUnits1 = libcellml::Units::create("myConcreteUnits1");
+    auto myImportedUnits1 = libcellml::Units::create("myImportedUnits1");
+    auto myConcreteUnits2 = libcellml::Units::create("myConcreteUnits2");
+    auto myImportedUnits2 = libcellml::Units::create("myImportedUnits2");
+
+    auto import1 = libcellml::ImportSource::create();
+    auto import2 = libcellml::ImportSource::create();
+
+    import1->setUrl("import1.cellml");
+    import2->setUrl("import2.cellml");
+
+    myImportedUnits1->setImportSource(import1);
+    myImportedUnits2->setImportSource(import2);
+
+    model->addUnits(myConcreteUnits1);
+    model->addUnits(myImportedUnits2);
+
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(1), model->importSourceCount());
+
+    // REPLACE concrete -> imported by reference.
+    EXPECT_TRUE(model->replaceUnits(myConcreteUnits1, myImportedUnits1));
+    EXPECT_EQ(size_t(2), model->unitsCount());
+    EXPECT_EQ(size_t(2), model->importSourceCount());
+    EXPECT_EQ(import2, model->importSource(0));
```

Comment:
Any particular reason for not having `EXPECT_EQ(import1, model->importSource(1));` as in the previous tests, or is that just to test the other import source and make sure that it is where you would expect it to be?

Reply:
Testing here to make sure the import source is where I expect.